spring-cloud-gcp-bigquery/src/main/java/org/springframework/cloud/gcp/bigquery/integration/outbound/BigQueryFileMessageHandler.java

```diff
@@ -0,0 +1,312 @@
+	}
+
+	/**
+	 * Sets the BigQuery table name to use.
+	 * @param tableName name of the BigQuery table
+	 */
+	public void setTableName(String tableName) {
+		this.tableNameExpression = new LiteralExpression(tableName);
+	}
+
+	/**
+	 * Sets the default {@link FormatOptions} to use for the handler.
+	 * The {@link FormatOptions} describe the type/format of data files being loaded.
+	 * @param formatOptions the format of the data file being loaded
+	 */
+	public void setFormatOptions(FormatOptions formatOptions) {
+		Assert.notNull(formatOptions, "Format options must not be null.");
+		this.formatOptionsExpression = new ValueExpression<>(formatOptions);
+	}
+
+	/**
+	 * Sets the SpEL {@link Expression} used to determine the {@link FormatOptions} for the handler.
+	 * @param formatOptionsExpression the SpEL expression used to evaluate the {@link FormatOptions}
+	 */
+	public void setFormatOptionsExpression(Expression formatOptionsExpression) {
+		Assert.notNull(formatOptionsExpression, "Format options expression cannot be null.");
+		this.formatOptionsExpression = formatOptionsExpression;
+	}
+
+	/**
+	 * Sets the {@link WriteDisposition} which specifies how data should be inserted into
+	 * BigQuery tables.
+	 * @param writeDisposition whether data should be appended or truncated to the BigQuery table
+	 */
+	public void setWriteDisposition(WriteDisposition writeDisposition) {
+		Assert.notNull(writeDisposition, "Write disposition must not be null.");
+		this.writeDisposition = writeDisposition;
+	}
+
+	/**
+	 * Sets whether BigQuery should attempt to autodetect the schema of the data when loading
+	 * data into an empty table for the first time.
+	 * @param autoDetectSchema whether data schema should be autodetected from the structure of
+	 * 		the data
+	 */
+	public void setAutoDetectSchema(boolean autoDetectSchema) {
+		this.autoDetectSchema = autoDetectSchema;
+	}
+
+	/**
+	 * Sets the {@link Duration} to wait for a file to be loaded into BigQuery before timing out
+	 * when waiting synchronously.
+	 * @param timeout the {@link Duration} timeout to wait for a file to load
+	 */
+	public void setTimeout(Duration timeout) {
+		Assert.notNull(timeout, "Timeout duration must not be null.");
+		this.timeout = timeout;
+	}
+
+	/**
+	 * Sets the {@link Duration} amount of time to wait between successive polls on the status of
+	 * a BigQuery job.
+	 * @param pollInterval the {@link Duration} poll interval for BigQuery job status polling
+	 */
+	public void setJobPollInterval(Duration pollInterval) {
+		Assert.notNull(pollInterval, "Poll interval duration must not be null.");
+		this.jobPollInterval = pollInterval;
+	}
+
+	/**
+	 * A {@code boolean} indicating if the {@link BigQueryFileMessageHandler} should synchronously
+	 * wait for each file to be successfully loaded to BigQuery.
+	 * @param sync whether {@link BigQueryFileMessageHandler} should wait synchronously for jobs to
+	 * 		complete
+	 */
+	public void setSync(boolean sync) {
```

Comment:
Sync is a bit of an overloaded abbreviation -- it could mean "synchronize" as in, "load changes to make the same" or "synchronous". Maybe "setBlocking".

Reply:
Well, this is what we use in other similar places 
So, for consistency I would insist to have it like this 😄

Reply:
Whoops. Withdrawing comment.